### PR TITLE
Remove some tests

### PR DESCRIPTION
**Pull request by emielver**
## Description & motivation
With the addition of anonymous tracking, we need to allow for the `domain_userid` and `network_userid` fields to be `NULL` where possible, so we need to remove tests that would fail when these fields are `NULL`. However, for the `User` module we still require `domain_userid` to be populated, and so we also need to introduce an additional filter at the `snowplow_web_user_sessions_this_run` level to enable the generation of user level data for all users who have opted in, and excluding those who have not. 

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have ~added~ removed tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

File: integration_tests/macros/edge_cases_to_ignore.sql [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {# Filters out edge cases in test data set which we haven't yet solved for #}
 {% macro edge_cases_to_ignore() %}
   user_id not in (
-    'stray page ping', -- Known unsolved issue https://github.com/snowplow/data-models/issues/92
-    'NULL domain_userid' -- Case when `domain_userid` is null but `domain_sessionid` is not null. Shouldn't happen. Will solve if it arises.
+    'stray page ping' -- Known unsolved issue https://github.com/snowplow/data-models/issues/92
```
Comment (rlh1994) [inline integration_tests/macros/edge_cases_to_ignore.sql:4]: Given the change we made to deal with this, can we actually remove this now?

Comment (emielver) [reply integration_tests/macros/edge_cases_to_ignore.sql:4]: Yes, will have to add the data to the affected tables but not a big deal, no edge cases to ignore anymore!

Comment (rlh1994) [reply integration_tests/macros/edge_cases_to_ignore.sql:4]: I think it might not actually need to be added - because in theory we were guessing with these before but now we know they get ignored (unless the whole session is being excluded?)

**Comment by emielver** [comment]
> @emielver Sorry this is after you've done it, but would we prefer to have those tests just have a warn severity instead? For people without anonymous tracking they might stil want these? https://docs.getdbt.com/reference/resource-configs/severity

On this note, I think removing the tests is the right call as we don't want to clog up unnecessarily with warnings when in fact nothing could be wrong. Consumers of the data will realise their tracking is setup incorrectly if their users table is empty, although I agree that's not the ideal way to find out. Should we instead add a test to see whether row_count of `derived.users` > 0? WDYT @rlh1994

**Comment by rlh1994** [comment]
> On this note, I think removing the tests is the right call as we don't want to clog up unnecessarily with warnings when in fact nothing could be wrong. Consumers of the data will realise their tracking is setup incorrectly if their users table is empty, although I agree that's not the ideal way to find out. Should we instead add a test to see whether row_count of derived.users > 0? WDYT @rlh1994

Yeah you're right. To be honest I think even that extra test is being a little assumptive about the users data (there may be some sites where they only do anonymous maybe?), and given our tests are more of a base to start from, I think just removing them is fine.

**Review by rlh1994 — APPROVED** [review]
```
  _                    _                                    _   _                           
 | |                  | |                                  | | | |                          
 | |      ___    ___  | | __ ___    __ _   ___    ___    __| | | |_  ___    _ __ ___    ___ 
 | |     / _ \  / _ \ | |/ // __|  / _` | / _ \  / _ \  / _` | | __|/ _ \  | '_ ` _ \  / _ \
 | |____| (_) || (_) ||   < \__ \ | (_| || (_) || (_) || (_| | | |_| (_) | | | | | | ||  __/
 |______|\___/  \___/ |_|\_\|___/  \__, | \___/  \___/  \__,_|  \__|\___/  |_| |_| |_| \___|
                                    __/ |                                                   
                                   |___/                                                    
```
